public/script.js

```diff
@@ -1870,24 +1870,26 @@ async function Generate(type, { automatic_trigger, force_name2, resolve, reject,
                         //anchorAndPersonality = "[Genre: roleplay chat][Tone: very long messages with descriptions]";
                         let personalityAndAnchor = [charPersonality, anchorTop].filter(x => x).join(' ');
                         if (personalityAndAnchor) {
-                            item += "[" + personalityAndAnchor + ']\n';
+                            item += "[" + personalityAndAnchor + "]\n";
                         }
                     }
-                    if (i === arrMes.length - 1 && coreChat.length > bottomAnchorThreshold && $.trim(item).substr(0, (name1 + ":").length) == name1 + ":" && !is_pygmalion) {//For add anchor in end
-                        item = item.substr(0, item.length - 1);
+                    if (i === arrMes.length - 1 && coreChat.length > bottomAnchorThreshold && item.trim().startsWith(name1 + ":") && !is_pygmalion) {//For add anchor in end
                         //chatString+=postAnchor+"\n";//"[Writing style: very long messages]\n";
-                        item = item + anchorBottom + "\n";
+                        if (anchorBottom) {
+                            item = item.replace(/\n$/, " ");
+                            item += anchorBottom + "\n";
+                        }
```

Comment:
Before:
```
You: This is a message.[Writing style: very long messages]
```

After:
```
You: This is a message. [Writing style: very long messages]
```

I was wondering if it really has any effect on model's output (LLaMA in my case), and this is the number of characters in a response I got (averaged from 10 regenerations over the same point of some natural looking conversation, token limit did not kick in):
* bottom anchor disabled: 82.6
* before this PR: 63.9
* after this PR: 82.2
* trying out bottom anchor on a separate line below user's message: 79.8
* trying out bottom anchor on a spearate line above user's message: 49.5

So my experience is that after this PR the bottom anchor has no effect, which is a noticeable improvement, because before it had a negative effect. ;)